src/dsp_tools/commands/xmllib/different_approaches/factory_option.py

```diff
@@ -0,0 +1,72 @@
+from __future__ import annotations
+
+import warnings
+from dataclasses import dataclass
+from typing import Any
+
+from lxml import etree
+
+from dsp_tools.commands.xmllib.value_checkers import is_integer
+from dsp_tools.models.custom_warnings import DspToolsUserWarning
+
+"""
+Advantage: 
+    Creation and usage separated
+    Res ID is included in the warning
+Disadvantage: 
+    Value classes can be instantiated without check (otherwise we would get duplicate warnings)
+    Each value also needs a dedicated factory.
+"""
+
+
+class IntValueFactory:
+    def create(
+        self,
+        value: int | str,
+        prop_name: str,
+        res_id: str,
```

Comment:
So the user has to provide the res_id manually? That's cumbersome, and people won't do it.

Reply:
The idea is that the users never create the classes themselves, but always use the function of the resource. This way only we have to provide that.